core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSequencerPartitionContainer.java

```diff
@@ -196,7 +196,6 @@ public void setRuns(Collection<Run> runs) {
   public Run getLastRun() {
     Run lastRun = null;
     for (Run thisRun : getRuns()) {
-      if (thisRun == null) continue;
       if (lastRun == null) {
         lastRun = thisRun;
       } else if (thisRun.getStatus() != null && thisRun.getStatus().getStartDate().after(lastRun.getStatus().getStartDate())) {
```

Comment:
Conditions for determining lastRun are still unbalanced. Both conditions should include the null check on `run.getStatus()`, or both should skip the null check. Status should never be null, and should be 1:1 with Run, so I'd say skip the null checks.